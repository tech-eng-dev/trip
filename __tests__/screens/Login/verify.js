import React from 'react';
import { shallow, configure } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
// imported as a connected component!
import Verify from '../../../src/screens/Login/verify';

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);
configure({ adapter: new Adapter() });

const initialState = {

};

describe('Login component renders correctly', () => {
  it('renders as expected', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(
      <Verify />,
      { context: { store } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
