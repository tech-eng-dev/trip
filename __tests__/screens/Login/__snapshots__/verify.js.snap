// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Verify
    fetchAllRecords={[Function]}
    fetchBlockData={[Function]}
    fetchMyRecords={[Function]}
    fetchUsers={[Function]}
    listenUserProfile={[Function]}
    logout={[Function]}
    myProfile={undefined}
    saveAllRecords={[Function]}
    saveBlockData={[Function]}
    saveMyRecords={[Function]}
    saveUserPhoto={[Function]}
    saveUserProfile={[Function]}
    saveUsers={[Function]}
    setFilterOption={[Function]}
    showToast={[Function]}
    welcome={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CustomHeader
          backgroundColor="#3DA0F1"
          left="ios-arrow-back"
          onPressLeft={[Function]}
          onPressRight={[Function]}
          right={undefined}
          title="Browse Trips"
        />,
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5FAFE",
              "flex": 1,
            }
          }
          testID="verify"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#5C6063",
                "fontSize": 20,
                "height": 200,
                "marginTop": 100,
                "paddingHorizontal": 30,
                "textAlign": "center",
              }
            }
          >
            The verification mail has been sent to the email you signed up. Please check your email address and click the url to continue. Click 
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "blue",
                }
              }
            >
              Confirm
            </Text>
             button if you verified.
          </Text>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-end",
                "paddingBottom": 50,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#5C6063",
                }
              }
            >
              Did you received verification email?
            </Text>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "marginBottom": 20,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#3DA0F1",
                    "textDecorationLine": "underline",
                  }
                }
              >
                Send again!
              </Text>
            </TouchableOpacity>
            <CustomButton
              backgroundColor="#6DBF53"
              onPress={[Function]}
              testID="verify-confirm-button"
              text="Confirm"
            />
          </View>
        </View>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColor": "#3DA0F1",
          "left": "ios-arrow-back",
          "onPressLeft": [Function],
          "onPressRight": [Function],
          "right": undefined,
          "title": "Browse Trips",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#5C6063",
                  "fontSize": 20,
                  "height": 200,
                  "marginTop": 100,
                  "paddingHorizontal": 30,
                  "textAlign": "center",
                }
              }
            >
              The verification mail has been sent to the email you signed up. Please check your email address and click the url to continue. Click 
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                Confirm
              </Text>
               button if you verified.
            </Text>,
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingBottom": 50,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#5C6063",
                  }
                }
              >
                Did you received verification email?
              </Text>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#3DA0F1",
                      "textDecorationLine": "underline",
                    }
                  }
                >
                  Send again!
                </Text>
              </TouchableOpacity>
              <CustomButton
                backgroundColor="#6DBF53"
                onPress={[Function]}
                testID="verify-confirm-button"
                text="Confirm"
              />
            </View>,
          ],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#F5FAFE",
            "flex": 1,
          },
          "testID": "verify",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "The verification mail has been sent to the email you signed up. Please check your email address and click the url to continue. Click ",
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                >
                  Confirm
                </Text>,
                " button if you verified.",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#5C6063",
                "fontSize": 20,
                "height": 200,
                "marginTop": 100,
                "paddingHorizontal": 30,
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              "The verification mail has been sent to the email you signed up. Please check your email address and click the url to continue. Click ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Confirm",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "blue",
                  },
                },
                "ref": null,
                "rendered": "Confirm",
                "type": [Function],
              },
              " button if you verified.",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#5C6063",
                    }
                  }
                >
                  Did you received verification email?
                </Text>,
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "marginBottom": 20,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#3DA0F1",
                        "textDecorationLine": "underline",
                      }
                    }
                  >
                    Send again!
                  </Text>
                </TouchableOpacity>,
                <CustomButton
                  backgroundColor="#6DBF53"
                  onPress={[Function]}
                  testID="verify-confirm-button"
                  text="Confirm"
                />,
              ],
              "style": Object {
                "flex": 1,
                "justifyContent": "flex-end",
                "paddingBottom": 50,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Did you received verification email?",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#5C6063",
                  },
                },
                "ref": null,
                "rendered": "Did you received verification email?",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeOpacity": 0.2,
                  "children": <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#3DA0F1",
                        "textDecorationLine": "underline",
                      }
                    }
                  >
                    Send again!
                  </Text>,
                  "onPress": [Function],
                  "style": Object {
                    "alignItems": "center",
                    "marginBottom": 20,
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Send again!",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "#3DA0F1",
                      "textDecorationLine": "underline",
                    },
                  },
                  "ref": null,
                  "rendered": "Send again!",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backgroundColor": "#6DBF53",
                  "onPress": [Function],
                  "testID": "verify-confirm-button",
                  "text": "Confirm",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CustomHeader
            backgroundColor="#3DA0F1"
            left="ios-arrow-back"
            onPressLeft={[Function]}
            onPressRight={[Function]}
            right={undefined}
            title="Browse Trips"
          />,
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F5FAFE",
                "flex": 1,
              }
            }
            testID="verify"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#5C6063",
                  "fontSize": 20,
                  "height": 200,
                  "marginTop": 100,
                  "paddingHorizontal": 30,
                  "textAlign": "center",
                }
              }
            >
              The verification mail has been sent to the email you signed up. Please check your email address and click the url to continue. Click 
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                Confirm
              </Text>
               button if you verified.
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingBottom": 50,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#5C6063",
                  }
                }
              >
                Did you received verification email?
              </Text>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#3DA0F1",
                      "textDecorationLine": "underline",
                    }
                  }
                >
                  Send again!
                </Text>
              </TouchableOpacity>
              <CustomButton
                backgroundColor="#6DBF53"
                onPress={[Function]}
                testID="verify-confirm-button"
                text="Confirm"
              />
            </View>
          </View>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "#3DA0F1",
            "left": "ios-arrow-back",
            "onPressLeft": [Function],
            "onPressRight": [Function],
            "right": undefined,
            "title": "Browse Trips",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#5C6063",
                    "fontSize": 20,
                    "height": 200,
                    "marginTop": 100,
                    "paddingHorizontal": 30,
                    "textAlign": "center",
                  }
                }
              >
                The verification mail has been sent to the email you signed up. Please check your email address and click the url to continue. Click 
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                >
                  Confirm
                </Text>
                 button if you verified.
              </Text>,
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "flex-end",
                    "paddingBottom": 50,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#5C6063",
                    }
                  }
                >
                  Did you received verification email?
                </Text>
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "marginBottom": 20,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#3DA0F1",
                        "textDecorationLine": "underline",
                      }
                    }
                  >
                    Send again!
                  </Text>
                </TouchableOpacity>
                <CustomButton
                  backgroundColor="#6DBF53"
                  onPress={[Function]}
                  testID="verify-confirm-button"
                  text="Confirm"
                />
              </View>,
            ],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "#F5FAFE",
              "flex": 1,
            },
            "testID": "verify",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "The verification mail has been sent to the email you signed up. Please check your email address and click the url to continue. Click ",
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "blue",
                      }
                    }
                  >
                    Confirm
                  </Text>,
                  " button if you verified.",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#5C6063",
                  "fontSize": 20,
                  "height": 200,
                  "marginTop": 100,
                  "paddingHorizontal": 30,
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Array [
                "The verification mail has been sent to the email you signed up. Please check your email address and click the url to continue. Click ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Confirm",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "blue",
                    },
                  },
                  "ref": null,
                  "rendered": "Confirm",
                  "type": [Function],
                },
                " button if you verified.",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#5C6063",
                      }
                    }
                  >
                    Did you received verification email?
                  </Text>,
                  <TouchableOpacity
                    activeOpacity={0.2}
                    onPress={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "marginBottom": 20,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "#3DA0F1",
                          "textDecorationLine": "underline",
                        }
                      }
                    >
                      Send again!
                    </Text>
                  </TouchableOpacity>,
                  <CustomButton
                    backgroundColor="#6DBF53"
                    onPress={[Function]}
                    testID="verify-confirm-button"
                    text="Confirm"
                  />,
                ],
                "style": Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingBottom": 50,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Did you received verification email?",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "#5C6063",
                    },
                  },
                  "ref": null,
                  "rendered": "Did you received verification email?",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activeOpacity": 0.2,
                    "children": <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "#3DA0F1",
                          "textDecorationLine": "underline",
                        }
                      }
                    >
                      Send again!
                    </Text>,
                    "onPress": [Function],
                    "style": Object {
                      "alignItems": "center",
                      "marginBottom": 20,
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "accessible": true,
                      "allowFontScaling": true,
                      "children": "Send again!",
                      "ellipsizeMode": "tail",
                      "style": Object {
                        "color": "#3DA0F1",
                        "textDecorationLine": "underline",
                      },
                    },
                    "ref": null,
                    "rendered": "Send again!",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "backgroundColor": "#6DBF53",
                    "onPress": [Function],
                    "testID": "verify-confirm-button",
                    "text": "Confirm",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
